auth/src/main/java/com/firebase/ui/auth/ui/accountlink/WelcomeBackIdpPrompt.java

```diff
@@ -127,83 +123,71 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     @Override
-    public void onClick(View view) {
-        next(mPrevIdpResponse);
-    }
-
-    @Override
-    public void onSuccess(IdpResponse idpResponse) {
-        next(idpResponse);
-    }
-
-    @Override
-    public void onFailure(Bundle extra) {
-        Toast.makeText(getApplicationContext(), "Error signing in", Toast.LENGTH_LONG).show();
-        finish(ResultCodes.CANCELED, IdpResponse.getErrorCodeIntent(ErrorCodes.UNKNOWN_ERROR));
-    }
-
-    private String getProviderIdFromIntent() {
-        return getIntent().getStringExtra(ExtraConstants.EXTRA_PROVIDER);
-    }
-
-    private String getEmailFromIntent() {
-        return getIntent().getStringExtra(ExtraConstants.EXTRA_EMAIL);
-    }
-
-    private void next(final IdpResponse newIdpResponse) {
-        if (newIdpResponse == null) {
+    public void onSuccess(final IdpResponse idpResponse) {
+        if (idpResponse == null) {
             return; // do nothing
         }
 
-        AuthCredential newCredential = AuthCredentialHelper.getAuthCredential(newIdpResponse);
+        AuthCredential newCredential = AuthCredentialHelper.getAuthCredential(idpResponse);
         if (newCredential == null) {
             Log.e(TAG, "No credential returned");
             finish(ResultCodes.CANCELED, IdpResponse.getErrorCodeIntent(ErrorCodes.UNKNOWN_ERROR));
             return;
         }
 
-        final FirebaseAuth firebaseAuth = mActivityHelper.getFirebaseAuth();
-        FirebaseUser currentUser = firebaseAuth.getCurrentUser();
-
+        FirebaseUser currentUser = mActivityHelper.getCurrentUser();
         if (currentUser == null) {
-            Task<AuthResult> authResultTask = firebaseAuth.signInWithCredential(newCredential);
-            authResultTask.addOnCompleteListener(new OnCompleteListener<AuthResult>() {
-                @Override
-                public void onComplete(@NonNull Task<AuthResult> task) {
-                    if (task.isSuccessful() && mPrevCredential != null) {
-                        FirebaseUser firebaseUser = task.getResult().getUser();
-                        firebaseUser.linkWithCredential(mPrevCredential);
-                        firebaseAuth.signOut();
-                        firebaseAuth
-                                .signInWithCredential(mPrevCredential)
-                                .addOnFailureListener(new TaskFailureLogger(
-                                        TAG, "Error signing in with previous credential"))
-                                .addOnCompleteListener(new FinishListener(newIdpResponse));
-                    } else {
-                        finish(ResultCodes.OK, IdpResponse.getIntent(newIdpResponse));
```

Comment:
BAD! (I think)
Here's the if: `if (task.isSuccessful() && mPrevCredential != null)`
What if the task fails!? We'll just finish with result ok? I'm pretty sure this was a mistake.